tasks/human_eval_129.rs

```diff
@@ -31,46 +33,42 @@ pub closed spec fn is_adjacent<const N: usize>(
 {
     &&& (0 <= coords.0 <= N - 1)
     &&& (0 <= coords.1 <= N - 1)
-    &&& (grid[coords.0][coords.1]) == m
-    &&& ((coords.0 < N - 1 && grid[coords.0 + 1][coords.1] == n) || (coords.0 > 0
-        && grid[coords.0][coords.1] == m && grid[coords.0 - 1][coords.1] == n) || (coords.1 < N - 1
-        && grid[coords.0][coords.1] == m && grid[coords.0][coords.1 + 1] == n) || (coords.1 > 0
-        && grid[coords.0][coords.1] == m && grid[coords.0][coords.1 - 1] == n))
+    &&& grid[coords.0][coords.1] == m
+    &&& ((coords.0 < N - 1 && grid[coords.0 + 1][coords.1] == n)
+         || (coords.0 > 0 && grid[coords.0 - 1][coords.1] == n)
+         || (coords.1 < N - 1 && grid[coords.0][coords.1 + 1] == n)
+         || (coords.1 > 0 && grid[coords.0][coords.1 - 1] == n))
```

Comment:
I may be overly fussy here, but I was imagining something like:
```
    &&& (   (coords.0 < N - 1 && grid[coords.0 + 1][coords.1] == n)
         || (coords.0 > 0     && grid[coords.0 - 1][coords.1] == n)
         || (coords.1 < N - 1 && grid[coords.0][coords.1 + 1] == n)
         || (coords.1 > 0     && grid[coords.0][coords.1 - 1] == n))
```
which makes it very transparent what's happening with the grid coordinates.

You could also consider starting the entire function body with a 
```
let (x, y) = coords;
...
```
so that the rest of the body isn't cluttered with `coords.0` and `coords.1`

Reply:
I fixed the formatting. Adding 
```
let (x, y) = coords;
...
```
causes the proof to fail and would then require changing a bunch of other things that makes things overall less readable.